tools/NonCopyableAnalyzer/NonCopyableAnalyzer.cs

```diff
@@ -0,0 +1,293 @@
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace NonCopyable
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class NonCopyableAnalyzer : DiagnosticAnalyzer
```

Comment:
Why can't we put it into a separate repository?

Maybe use as a submodule if you don't want to reference it as a package.

Reply:
It's currently buggy. Once evrything works, we can move the changes to your repo again.